File: src/hyperliquid_python/services/parse_webhook.py
from models.webhook import WebhookPayload

def parse_webhook(payload: dict) -> WebhookPayload:
    """
    Parse and normalize webhook payload
    Removes "USDT" suffix from symbol names to match HyperLiquid format
    """
    symbol = payload.get("symbol", "")
    if symbol.endswith("USDT"):
        payload["symbol"] = symbol[:-4]
    
    return WebhookPayload(**payload)

File: src/hyperLiquid_ts_deprecated/services/parseWebhook.ts
import { WebhookPayload } from "../models/webhook";

/**
 * Parse and normalize webhook payload
 * Removes "USDT" suffix from symbol names to match HyperLiquid format
 * @param payload - Raw webhook payload from TradingView
 * @returns Normalized webhook payload with cleaned symbol name
 */
export function parseWebhook(payload: WebhookPayload): WebhookPayload {
    return {
        ...payload,
        symbol: payload.symbol.replace("USDT", "")
        // symbol: payload.symbol.replace("USDT", "").replace("USD", "")
    };
}
